### DISCUSSION TITLE
Markdown parser

### DISCUSSION THREAD

**Discussion by jcubic** [peggyjs/peggy#325] [General]
Does anyone have any experience with Markdown parsers? Does such a parser for PEG.js or Peggy.js exist? Should be simple. I'm worrying about links, it may be difficult to make them right if they contain brackets and parentheses.

I've planned to create an extension with Peggy that implement markdown for my library jQuery Terminal. But I think it would be much more fun to create a tutorial on how to do that. Of course, full code will be available. It will benefit both projects. But I don't plan to support all of HTML like a normal parser. Only tokens that are in Markdown itself.

Just wanted to share the idea. It's tracked here https://github.com/jcubic/jquery.terminal/issues/827

**Comment by hildjj** [comment]
I have the very start of a really bad purpose-specific one in peggy-test, [here](https://github.com/hildjj/peggy-test/blob/main/src/test_md.peggy).  I'd be willing to collaborate on a real one.  That might be a big enough thing that it should be its own project?

**Reply by jcubic** [reply]
I've created a basic parser for my needs: [Markdown jQuery Terminal formatter](https://github.com/jcubic/jquery.terminal/wiki/Formatting-and-Syntax-Highlighting#markdown-parser)
